docs/source/topics/data_types.rst

```diff
@@ -0,0 +1,508 @@
+.. _topics:data_types:
+
+**********
+Data types
+**********
+
+AiiDA already ships with a number of useful data types.
+This section details the most common, and some handy features/functionalities to work with them.
+
+.. _topics:data_types:core:
+
+Core data types
+===============
+
+Here follows a short summary of common data types already provided with AiiDA.
+This list is not complete, see also inside :py:mod:`aiida.orm.nodes.data` for the list of all available plugins.
+
+We also mention, when relevant, what is stored in the database (as attributes, so that it can be easily queried e.g. with the :ref:`QueryBuilder <how-to:data:find>`) and what is stored in the file repository (providing access to the file contents, but not efficiently queryable: this is useful for e.g. big data files that don't need to be queried for).
+
+For all data types, you can follow the link to the data class to read more about the methods provided, how to access them, and so on.
+
+If you need to work with some specific type of data, first check the list of data types/plugins below, and if you don't find what you need, give a look to :ref:`how to write a new data plugin <how-to:data:plugin:create>`.
+
+.. _topics:data_types:core:base:
+
+Base types
+----------
+
+In the :py:mod:`aiida.orm.nodes.data.base` module there are a number of useful classes that wrap base python datatypes (like :py:class:`~aiida.orm.nodes.data.int.Int`, :py:class:`~aiida.orm.nodes.data.float.Float`, :py:class:`~aiida.orm.nodes.data.str.Str`, ...).
+These are automatically loaded with the verdi shell, and also directly exposed from :py:mod:`aiida.orm`.
+These classes are particularly useful when you need to provide a single parameter to e.g. a :py:class:`~aiida.engine.processes.functions.workfunction`.
+
+Each of these classes can most often be used transparently (e.g. you can sum two :py:class:`~aiida.orm.nodes.data.int.Int` objects, etc.). If you need to access the bare value and not the whole AiiDA class, use the ``.value`` property.
```

Comment:
I would add a warning below, saying something along the lines:

Warning! While this is convenient if e.g. you need to do simple manipulations, like multiplying two numbers, be very careful of not passing such nodes instead of the corresponding python values to libraries that perform heavy computations with them: in fact, any operation on the value would be replaced with an operation creating new AiiDA nodes, that however can be orders of magnitude slower. In this case, remember to pass the `node.value` to the mathematical function instead. One can reference #3320